src/main/webapp/app/exercises/shared/team/teams-import-dialog/teams-import-dialog.component.ts

```diff
@@ -239,11 +287,56 @@ export class TeamsImportDialogComponent implements OnInit, OnDestroy {
         if (this.isSubmitDisabled) {
             return;
         }
-        this.isImporting = true;
-        this.teamService.importTeamsFromSourceExercise(this.exercise, this.sourceExercise, this.importStrategy!).subscribe(
-            (res) => this.onSaveSuccess(res),
-            () => this.onSaveError(),
-        );
+        if (this.showImportFromExercise) {
+            this.isImporting = true;
+            this.teamService.importTeamsFromSourceExercise(this.exercise, this.sourceExercise!, this.importStrategy!).subscribe(
+                (res) => this.onSaveSuccess(res),
+                (error) => this.onSaveError(error),
+            );
+        } else if (this.sourceTeams) {
+            this.notFoundRegistrationNumbers = [];
+            this.teamService.importTeams(this.exercise, this.sourceTeams, this.importStrategy!).subscribe(
+                (res) => this.onSaveSuccess(res),
+                (error) => this.onSaveError(error),
+            );
+        }
+    }
+
+    /**
+     * Update source teams to given teams
+     * @param {Team[]} fileTeams - Teams which its students only have visible registration number
+     */
+    onTeamsChanged(fileTeams: Team[]) {
+        this.initImportStrategy();
+        this.sourceTeams = fileTeams;
+        this.studentRegistrationNumbersAlreadyExistingInOtherTeams = [];
+        this.notFoundRegistrationNumbers = [];
+        const students: User[] = flatMap(fileTeams, (fileTeam) => fileTeam.students ?? []);
+        const loginHistogram = students.reduce(function (map, student) {
+            const login: string | undefined = student.login;
+            if (login) {
+                if (map[login]) {
+                    map[login] += 1;
+                } else {
+                    map[login] = 1;
+                }
+            }
+            return map;
+        }, {});
+        const registrationNumberHistogram = students.reduce(function (map, student) {
+            const registrationNumber: string | undefined = student.visibleRegistrationNumber;
+            if (registrationNumber) {
+                if (map[registrationNumber]) {
+                    map[registrationNumber] += 1;
+                } else {
+                    map[registrationNumber] = 1;
+                }
+            }
+            return map;
+        }, {});
+        this.studentLoginsAlreadyExistingInOtherTeams = Object.keys(loginHistogram).filter((key) => loginHistogram[key] > 1);
```

Comment:
we should use actual Maps in Typescript and not JavaScript objects with Object.keys